File: 백준/Silver/1003. 피보나치 함수/피보나치 함수.py
import sys
input = sys.stdin.readline

test_case = int(input().rstrip("\n"))
dp = [[1,0]]
dp_calced = 0
for i in range(test_case):
    num = int(input().rstrip("\n"))
    while(dp_calced < num):
        dp.append([dp[dp_calced][1], dp[dp_calced][0]+dp[dp_calced][1]])
        dp_calced+=1
    sys.stdout.write(f"{dp[num][0]} {dp[num][1]}\n")

File: 백준/Silver/1003. 피보나치 함수/피보나치 함수.cc
#include<iostream>
#include<vector>
using namespace std;
int main(void) {
	vector<vector<long long int>> dp(1);
	dp[0] = { 1,0 };
	int test_case, dp_calced=0;
	cin >> test_case;

	while (test_case--) {
		int num;
		cin >> num;
		while (dp_calced < num) {
			dp.push_back({ dp[dp_calced][1], dp[dp_calced][0] + dp[dp_calced][1] });
			dp_calced++;
		}
		cout << dp[num][0] << " " << dp[num][1] << endl;
	}
	return 0;
}